src/plugins/embeddable/public/lib/embeddables/embeddable_factory.ts

```diff
@@ -115,4 +117,13 @@ export interface EmbeddableFactory<
     initialInput: TEmbeddableInput,
     parent?: IContainer
   ): Promise<TEmbeddable | ErrorEmbeddable | undefined>;
+
+  migrate(input: Serializable): EmbeddableInput;
+  extract: (
+    input: EmbeddableInput
+  ) => {
+    input: EmbeddableInput;
+    references: SavedObjectReference[];
+  };
+  inject: (input: EmbeddableInput, references: SavedObjectReference[]) => EmbeddableInput;
```

Comment:
I know we talked about this earlier, but I wonder if we should be providing a base set of interfaces for migrate/extract/inject functions that each plugin would need to implement or extend?

e.g.
```ts
// somewhere in core?
interface Persistable<State extends Serializable> {
  migrate: (state: Serializable) => State; 
  extract: (state: State) => { state: State, references: SavedObjectReference[] };
  inject: (state: State, references: SavedObjectReference[]) => State;
}

// in embeddables plugin
interface EmbeddableInstanceConfiguration extends Persistable<EmbeddableInput> {
  ...extract/inject/migrate are all typed
}
```